index.js

```diff
@@ -1,58 +1,108 @@
 // implement your API here
 const express = require("express");
 const server = express();
-const db = require("./data/db.js");
 const port = 8000;
+const db = require("./data/db.js");
+const cors = require("cors");
+
+server.use(express.json());
+server.use(cors());
+
+const sendError = (res, status, msg) => {
+  res.status(status).json({ error: msg });
+  return;
+};
+
+const sendSuccess = (res, status = 200, data, msg) => {
+  res.status(status).json({ data, message: msg });
+  return;
+};
 
 server.get("/", (req, res) => {
   res.send("Hello World, from Node.js!!!!!");
 });
 
 server.post("/api/users", (req, res) => {
-  res.status(201).json({ url: "/api/users", operation: "POST" });
+  const { name, bio } = req.body;
+  if (!name || !bio) {
+    sendError(res, 400, "Provide name and bio");
+    return;
+  }
+  db.insert({ name, bio })
+    .then(response => {
+      sendSuccess(res, 201, response, "User successfully added!");
+    })
+    .catch(() => {
+      sendError(
+        res,
+        500,
+        "There was an error while saving the user to the database"
+      );
+      return;
+    });
 });
 
 server.get("/api/users", (req, res) => {
   db.find()
     .then(users => {
-      res.json(users);
+      // res.json(users);
+      sendSuccess(res, 200, users, "Successfully retrieved list of users!");
     })
-    .catch(err => {
-      res.status(500).json({
-        error: err,
-        msg: "he users information could not be retrieved."
-      });
+    .catch(() => {
+      sendError(res, 500, "The users information could not be retrieved.");
     });
 });
 
 server.get("/api/users/:id", (req, res) => {
   let id = req.params.id;
   db.findById(id)
     .then(user => {
-      res.json(user);
+      if (user === undefined) {
+        sendError(res, 404, `The user with ID: ${id} does not exist.`);
+      }
+      sendSuccess(res, 200, user, `Successfully retrieved user with ID: ${id}`);
     })
-    .catch(err => {
-      res.status(404).json({
-        error: err,
-        msg: "The user with the specified ID does not exist."
-      });
+    .catch(() => {
+      sendError(res, 500, "The user's information could not be retrieved.");
     });
 });
 
 server.delete("/api/users/:id", (req, res) => {
   let id = req.params.id;
   db.remove(id)
-    .then(res.send(204))
-    .catch(err => {
-      res.status(404).json({
-        error: err,
-        msg: "The user with the specified ID does not exist."
-      });
+    .then(response => {
+      if (response === 0) {
```

Comment:
response will just about never be 0 unless that is the id number that is sent in. You should either use `!response` or check if the length of the response is 0 here.